File: Baekjoon/S5/C#/S5_요세푸스 문제 0.cs
https://www.acmicpc.net/problem/11866

using System;
using System.Collections.Generic;
using System.Text;

namespace Baekjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
            int n = arr[0];
            int k = arr[1];

            Queue<int> q = new Queue<int>();
            for (int i = 1; i <= n; i++)
                q.Enqueue(i);

            StringBuilder sb = new StringBuilder();
            sb.Append("<");
            while (q.Count > 0)
            {
                for (int i = 1; i < k; i++)
                    q.Enqueue(q.Dequeue());
                sb.Append(q.Dequeue());
                sb.Append((q.Count > 0) ? ", " : ">");
            }
            Console.Write(sb);
        }
    }
}

File: Baekjoon/S5/JAVA/S5_요세푸스 문제 0.java
//https://www.acmicpc.net/problem/11866

import java.io.*;
import java.util.*;

public class Main{
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        String[] arr = br.readLine().split(" ");
        int n = Integer.parseInt(arr[0]);
        int k = Integer.parseInt(arr[1]);
        
        Queue<Integer> q = new LinkedList<Integer>();
        for(int i = 1; i <= n; i++)
            q.add(i);
        
        StringBuilder sb = new StringBuilder();
        sb.append("<");
        while(q.size() > 0){
            for(int i = 0; i < k-1; i++)
                q.add(q.poll());
                
            sb.append(q.poll());
            sb.append((q.size() > 0)? ", ":">");
        }
        
        System.out.print(sb);
        br.close();
    }
}